pkg/storage/provider_list_actions_test.go

```diff
@@ -0,0 +1,144 @@
+package storage_test
+
+import (
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/fixtures"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/fixtures/testusers"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/validate"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/randomizer"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/storage"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/storage/internal/testabilities"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListActions_HappyPath(t *testing.T) {
+	// Given:
+	ctx := t.Context()
+	given, cleanup := testabilities.Given(t)
+	defer cleanup()
+
+	activeStorage := given.Provider().WithRandomizer(randomizer.NewTestRandomizer()).GORM()
+
+	// and:
+	given.Faucet(activeStorage, testusers.Alice).TopUp(100_000)
+
+	// and:
+	_, err := activeStorage.CreateAction(ctx, testusers.Alice.AuthID(), fixtures.DefaultValidCreateActionArgs())
+	require.NoError(t, err)
+
+	// When:
+	args := wdk.ListActionsArgs{
+		Limit:            10,
+		Offset:           0,
+		IncludeAllLabels: false, // "any=false" or "all=true"
+		IncludeLabels:    true,
+		IncludeOutputs:   true,
+		IncludeInputs:    true,
```

Comment:
we should have separated tests for each of available flags and also for limit and offset.

But you can add it in another PR, if you want to merge this one.